source/agora/consensus/data/Block.d

```diff
@@ -110,6 +118,15 @@ public struct BlockHeader
         this.merkle_root = Hash(dg(Hash.sizeof));
         deserializePart(this.validators, dg);
         deserializePart(this.signature, dg);
+
+        size_t enrollments_size;
+        deserializePart(enrollments_size, dg);
+        foreach (_; 0 .. enrollments_size)
+        {
+            Enrollment enrollment;
+            deserializePart(enrollment , dg);
+            this.enrollments ~= enrollment;
```

Comment:
This will perform one allocation per one allocation. It's *very* inefficient.
Instead, since you already know the size of the whole array, you can do the following:
```D
 size_t enrollments_size;
deserializePart(enrollments_size, dg); 
this.enrollments.length = enrollments_size; // Only do one allocation
foreach (ref val; this.enrollments)
  deserializePart(val, dg);
```

Note that if `this.enrollments` is `null`, all those statements are equals:
```D
size_t len = 42; // Not this, of course :)
this.enrollments.length = len;
this.enrollments = new typeof(this.enrollments)(len);
this.enrollments = new Enrollment[](len);
```

While this is more efficient, an array allocation leads to a default-initialization of every elements, and for this reason instead of setting the length / allocation with the GC, a common optimization is to use https://dlang.org/phobos/std_array.html#uninitializedArray